server/api/collections.js

```diff
@@ -51,6 +54,29 @@ router.post("collections.create", auth(), async (ctx) => {
   const user = ctx.state.user;
   authorize(user, "create", Collection);
 
+  const collections = await Collection.findAll({
+    where: { teamId: user.teamId, deletedAt: null },
+    attributes: ["id", "index", "updatedAt"],
+  });
+
+  if (index) {
+    const allowedASCII = new RegExp(/^[\x21-\x7E]+$/);
+    if (!allowedASCII.test(index)) {
+      throw new ValidationError(
+        "Index characters must be between x21 to x7E ASCII"
+      );
+    }
+  } else {
+    collections.sort((a, b) => {
```

Comment:
I had to sort the collections manually because `order by` wasn't returning the collections in the order presented in the app.

Reply:
Do you mean the postgres order wasn't working as expected? I don't think its feasible to just query all the collections and sort them in js, I can look into the postgres ordering if that's the problem.

Reply:
Sorry to put you through this work, but I think you should take a look as well. This was my query.

```js
  const collections = await Collection.findAll({
    where: { teamId: user.teamId, deletedAt: null },
    attributes: ["id", "index", "updatedAt"],
    order: [["index", "DESC"]],
    limit: 1,
  });
```

Reply:
Try the following:

`order: ['index COLLATE "C"', "ASC"]],`

Using collate makes postgres sort by the byte-order ([ref](https://dba.stackexchange.com/questions/94887/what-is-the-impact-of-lc-ctype-on-a-postgresql-database)), which is what we want in this case. I'd switch it round so that new collections go to the top – they could be lost easily at the bottom, this will also match the behavior of new docs.

Reply:
This was very informative, Thanks Tom! 

Yeah sure, the current implementation was inspired by Trello's way of creating cards. Now, Let's make the new collections go to the top.